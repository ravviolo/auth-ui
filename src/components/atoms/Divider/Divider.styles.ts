import styled from 'styled-components';

import { colors } from 'theme';

export const Container = styled.div`
  position: relative;
  height: 10px;
`;
export const Line = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  height: 1px;
  background-color: ${colors.textLight};
`;

export const Badge = styled.span`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #fff;
  color: ${colors.textLight};
  padding: 4px;
  font-size: 0.8rem;
  text-transform: uppercase;
  border-radius: 5px;
  border: 1px solid ${colors.textLight};
  font-family: 'Roboto', sans-serif;
`;
