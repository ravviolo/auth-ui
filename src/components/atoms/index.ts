export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
