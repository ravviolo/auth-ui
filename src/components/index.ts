export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
