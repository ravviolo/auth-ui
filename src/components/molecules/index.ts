export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
