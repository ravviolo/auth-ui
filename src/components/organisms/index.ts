export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
