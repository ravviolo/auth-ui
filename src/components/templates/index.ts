export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
