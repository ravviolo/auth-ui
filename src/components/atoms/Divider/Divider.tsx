import * as S from './Divider.styles';

export interface Props {
  label?: string;
  testId: string;
}

export const Divider = ({ label, testId }: Props) => {
  return (
    <S.Container data-testid={testId}>
      <S.Line />
      {label && <S.Badge>{label}</S.Badge>}
    </S.Container>
  );
};
