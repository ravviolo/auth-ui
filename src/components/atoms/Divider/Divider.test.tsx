import { render, screen, within } from '@testing-library/react';
import { create } from 'react-test-renderer';

import { Divider } from 'components/atoms';

import { Props as DividerProps } from './Divider';

describe('Divider', () => {
  const props: DividerProps = {
    testId: 'test-divider',
  };

  it('renders correctly without badge', () => {
    render(<Divider {...props} />);

    const divider = screen.getByTestId('test-divider');

    expect(divider).toBeInTheDocument();

    const badgeText = within(divider).queryByText('Badge');

    expect(badgeText).not.toBeInTheDocument();
  });

  it('renders correctly with badge', () => {
    render(<Divider {...props} label="Badge" />);

    const divider = screen.getByTestId('test-divider');

    expect(divider).toBeInTheDocument();

    const badgeText = within(divider).getByText('Badge');

    expect(badgeText).toBeInTheDocument();
  });

  it('matches snapshot with badge', () => {
    const tree = create(<Divider {...props} label="Badge" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot without badge', () => {
    const tree = create(<Divider {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
