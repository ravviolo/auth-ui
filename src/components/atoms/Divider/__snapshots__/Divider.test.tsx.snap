// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider matches snapshot with badge 1`] = `
<div
  className="sc-dkzDqf fBSlBa"
  data-testid="test-divider"
>
  <div
    className="sc-hKMtZM cJojKy"
  />
  <span
    className="sc-eCYdqJ fGmLST"
  >
    Badge
  </span>
</div>
`;

exports[`Divider matches snapshot without badge 1`] = `
<div
  className="sc-dkzDqf fBSlBa"
  data-testid="test-divider"
>
  <div
    className="sc-hKMtZM cJojKy"
  />
</div>
`;
